File: backend/src/main/java/com/optimiza/clickbarber/repository/BarbeariaRepository.java
package com.optimiza.clickbarber.repository;

import com.optimiza.clickbarber.model.Barbearia;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface BarbeariaRepository extends JpaRepository<Barbearia, Long> {

    Optional<Barbearia> findByIdExterno(UUID idExterno);

    Optional<Barbearia> findByUsuarioId(Long usuarioId);

    List<Barbearia> findByNomeContainingIgnoreCase(String nome);
}

File: backend/src/main/java/com/optimiza/clickbarber/repository/BarbeariaRepository.kt
package com.optimiza.clickbarber.repository

import com.optimiza.clickbarber.model.Barbearia
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface BarbeariaRepository : JpaRepository<Barbearia, Long> {
    fun findByUsuarioId(usuarioId: UUID?): Optional<Barbearia>

    @Query("SELECT b FROM Barbearia b WHERE nome like %:nome%")
    fun findByNome(nome: String): List<Barbearia>
}